src/Homie/Uptime.cpp

```diff
@@ -4,15 +4,21 @@ using namespace HomieInternals;
 
 Uptime::Uptime()
 : _seconds(0)
-, _lastTick(0) {
+, _lastTick(0)
+, _rolloverCount(0)   {
 }
 
 void Uptime::update() {
   uint32_t now = millis();
-  _seconds += (now - _lastTick) / 1000UL;
+  
+  if (now < _lastTick) {
+    _rolloverCount++;
+  }
+  
   _lastTick = now;
+  _seconds =  (0xFFFFFFFF / 1000 ) * _rolloverCount + (_lastTick / 1000);
```

Comment:
How about:

```
_seconds = (((uint64_t) rollover_count << 32) + last_tick) / 1000
```

I would store the uint64_t value of `((uint64_t) rollover_count << 32)` rather than the uint32_t `rollover_count`, since I think it needs less memory overall.